7_Classes/singleton_class.py

```diff
@@ -0,0 +1,23 @@
+""" object:
+    The base class of the class hierarchy.
+
+    When called, it accepts no arguments and returns a new featureless
+    instance that has no instance attributes and cannot be given any.
+    """
+
+
+class SingletonClass(object):
+    def __new__(cls):
+        if not hasattr(cls, 'instance'):
+            cls.instance = super(SingletonClass, cls).__new__(cls)
+        return cls.instance
```

Comment:
more advanced task - I need a singleton class that returns the same instance only if init args match

```
a1 = Singleton('AAA')
a2 = Singleton('AAA')
a1 is a2 # must be true
b1 = Singleton('BBB')
a1 is b1 # must be false
```